### ISSUE TITLE
Keras model Can't pickle _thread.RLock objects

### ISSUE THREAD

**Issue by gootacatchitall** [tensorflow/tensorflow#56466]
`from __future__ import absolute_import
from __future__ import print_function

import numpy as np
from keras.applications.vgg16 import VGG16
from keras.layers import *
from keras.models import Model, load_model
from keras.optimizers import SGD
from keras.preprocessing.image import load_img, img_to_array
import tensorflow as tf
from keras import backend as K

config = tf.ConfigProto()
config.gpu_options.allow_growth = True
sess = tf.Session(config=config)
K.set_session(sess)

def convnet_model_():
    vgg_model = VGG16(weights=None, include_top=False)
    x = vgg_model.output
    x = GlobalAveragePooling2D()(x)
    x = Dense(4096, activation='relu')(x)
    x = Dropout(0.6)(x)
    x = Dense(4096, activation='relu')(x)
    x = Dropout(0.6)(x)
    x = Lambda(lambda x_: K.l2_normalize(x, axis=1))(x)
    convnet_model = Model(inputs=vgg_model.input, outputs=x)
    return convnet_model

def deep_rank_model():
    convnet_model = convnet_model_()
    first_input = Input(shape=(224, 224, 3))
    first_conv = Conv2D(96, kernel_size=(8, 8), strides=(16, 16), padding='same')(first_input)
    first_max = MaxPool2D(pool_size=(3, 3), strides=(4, 4), padding='same')(first_conv)
    first_max = Flatten()(first_max)
    first_max = Lambda(lambda x: K.l2_normalize(x, axis=1))(first_max)

    second_input = Input(shape=(224, 224, 3))
    second_conv = Conv2D(96, kernel_size=(8, 8), strides=(32, 32), padding='same')(second_input)
    second_max = MaxPool2D(pool_size=(7, 7), strides=(2, 2), padding='same')(second_conv)
    second_max = Flatten()(second_max)
    second_max = Lambda(lambda x: K.l2_normalize(x, axis=1))(second_max)

    merge_one = concatenate([first_max, second_max])

    merge_two = concatenate([merge_one, convnet_model.output])
    emb = Dense(4096)(merge_two)
    l2_norm_final = Lambda(lambda x: K.l2_normalize(x, axis=1))(emb)

    final_model = Model(inputs=[first_input, second_input, convnet_model.input], outputs=l2_norm_final)

    return final_model

As can see this code having lambda, which may fail when it try to serialize an unserializable object.
My question is what should I add to my lambda expression to avoid this.
i must save the model in term of using in web.
`

**Comment by gootacatchitall** [comment]
@tilakrayal I tried ur method but come out with new bug...
![image](https://user-images.githubusercontent.com/50090044/174036284-a0bb5e9b-cff0-4d70-b185-6e947e373fb4.png)

I am using tf 1.x version because this bug [#2458](https://github.com/matterport/Mask_RCNN/issues/2458)

Btw,I just put my gist here so u can have full view on the code.
https://colab.research.google.com/drive/1CEogpAk3MAbF-3uDXVKiUd2h9rQXkLWU?usp=sharing

**Comment by gootacatchitall** [comment]
> @gootacatchitall, From the provided code, we can assume that you are using tensorflow 1.x which is not actively supported. I tried to convert the code for the 2.x version and I was able to execute without any issues. Kindly find the gist [here](https://colab.research.google.com/gist/tilakrayal/779e4d5e395022e39dc82d214384d841/56466.ipynb) and please update to the 2.x version and let us know if you are facing the same issue. Thank you!

I can't use tensor 2.x version as my code cant support it

**Comment by tilakrayal** [comment]
@gootacatchitall,
Unfortunately TensorFlow 1.x is not actively supported. Could you please update TensorFlow to the latest stable version v2.9.0 and check if you are facing the same issue. 
Also it would be difficult for us to debug the issue with 1.x code. Kindly provide the version 2.x compatible code. Thank you!

**Comment by gootacatchitall** [comment]
> @gootacatchitall, Unfortunately TensorFlow 1.x is not actively supported. Could you please update TensorFlow to the latest stable version v2.9.0 and check if you are facing the same issue. Also it would be difficult for us to debug the issue with 1.x code. Kindly provide the version 2.x compatible code. Thank you!


![image](https://user-images.githubusercontent.com/50090044/174546231-15fb77db-6a42-48f2-b6c7-e6ba61c85537.png)

i got a new error when trying to use tensorflow 2.x...it seems like I can put a layer with keras.k.l2.normalize

Kindly find the gist as below
https://colab.research.google.com/github/gootacatchitall/WID3003-NeuralComputing/blob/main/56466.ipynb?authuser=2

**Comment by tilakrayal** [comment]
@gootacatchitall,
Thanks for opening this issue. Development of keras moved to another [repository](https://github.com/keras-team/keras/issues). 
 
 
Could you please post this issue on keras-team/keras [repo](https://github.com/keras-team/keras/issues).
To know more please refer:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!

**Comment by roccopietrini** [comment]
The code posted in the colab by @tilakrayal does not work, there are only definition there.. as soon as you call them you have the same error encountered by @gootacatchitall.
Look [here](https://github.com/USCDataScience/Image-Similarity-Deep-Ranking/issues/4#issuecomment-1171287956) for the solution.
